pkg/controllers/managementapi/usercontrollers/usercontroller.go

```diff
@@ -94,10 +102,65 @@ func (u *userControllersController) sync(key string, cluster *v3.Cluster) (runti
 	if key == all {
 		return nil, u.setPeers(nil)
 	}
+
+	// Check if the cluster has been upgraded to a major or minor version and restart the controllers.
+	if cluster != nil && cluster.Status.Version != nil {
+		var err error
+		currentVersion, err := getCurrentClusterControllersVersion(cluster)
+		// If the version is not found on the annotation, set it and update the cluster object.
+		if err != nil {
+			if cluster, err = u.saveClusterControllersVersionAnnotation(cluster, cluster.Status.Version.String()); err != nil {
+				return nil, err
+			}
+			return cluster, nil
+		}
+
+		newVersion, err := version.ParseSemantic(cluster.Status.Version.String())
+		if err != nil {
+			return cluster, nil
```

Comment:
I don't think ignoring the error here is the right move. If we can't parse the version of the cluster, we won't restart the controller. This could result in us failing to restart on certain clusters where we need to (i.e. k3s and RKE/RKE2 have odd version semantics).

Since these are called on any update of the cluster and this won't just fix itself, I think logging the issue with a level of error would be a good idea.